account_invoice_pricelist/model/account_invoice_line.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+# © 2015-2016 GRAP <http://www.grap.coop>.
+# © 2017 Therp BV <http://therp.nl>.
+# License AGPL-3.0 or later <http://www.gnu.org/licenses/agpl.html>.
+from openerp import models, api
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = 'account.invoice.line'
+
+    @api.multi
+    def product_id_change(
+            self, product_id, uom_id, qty=0, name='', type='out_invoice',
+            partner_id=False, fposition_id=False, price_unit=False,
+            currency_id=False, company_id=None):
+        """Make sure pricelist is used on product.
+
+        NB: On original method argument product_id is confusingly called
```

Comment:
every argument is a keyword argument from the point of view of the interpreter:

``python
>>> def demo(demo0):
...     print demo0
... 
>>> demo('42')
42
>>> demo(demo0='42')
42
>>> demo(demo0_id='42')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: demo() got an unexpected keyword argument 'demo0_id'
```